src/content/docs/en/pages/guides/marketplace/integrations/rate-limiting-with-penalty/rate-limiting-with-penalty.mdx

```diff
@@ -0,0 +1,192 @@
+---
+title: How to install Rate Limiting with Penalty through Azion Marketplace
+description: This integration allows you to control of incoming traffic right at the edge and protect your applications.
+meta_tags: marketplace, edge computing, rate limiting
+namespace: docs_guides_rate_limiting_penalty
+permalink: /documentation/products/guides/rate-limiting-with-penalty/
+---
+
+Azion **Rate Limiting with Penalty** allows you to control of incoming traffic right at the edge of the network, avoiding bottlenecks, managing traffic spikes, and protecting your applications from potential threats such as DDoS, fuzzing, or brute force attacks. 
+
+With this integration, based on an edge function, you can define:
+
+- **Limiting requests** and **window limit**, including specific rate limit windows for different periods of the day.
+- A **penalty** configuration that analyzes the validity of each request.
+  - If the request isn't valid, it blocks the request and returns the user a `403 Forbidden` status code. If the request is valid, it counts the request and, in case it reaches the defined rate limit, it interrupts the request and returns a `429 Too Many Requests` status code.
+
+It's integrated with the `upstash/ratelimit` library and provides a global accounting of the rate limit, summing up all the requests received across the entire network, instead of counting separately for each edge location.
```

Comment:
```suggestion
Rate Limiting with Penalty is integrated with the `upstash/ratelimit` library and provides a global accounting of the rate limit, summing up all the requests received across the entire network, instead of counting separately for each edge location.
```